site/cypress/e2e/pdfs.spec.js

```diff
@@ -0,0 +1,105 @@
+import {getCurrentSemester} from '../support/utils.js';
+
+function switch_settings(setting, pdf_type) {
+    cy.visit(`/courses/${getCurrentSemester()}/sample`);
+    cy.login('instructor');
+    cy.get(`a[href*="/sample/gradeable/${pdf_type}/update"]`).click();
+    cy.get('#page_3_nav').click();
+    cy.get('#minimum_grading_group option:selected').invoke('text').as('minimumGradingGroup');
+    cy.get('#minimum_grading_group').select(setting);
+    cy.get('#minimum_grading_group option:selected').should('have.text', setting);
+    cy.get('input[name="grader_assignment_method"]:checked').as('graderAssignmentMethod');
+    cy.get('input[name="grader_assignment_method"][value="1"]').check();
+    cy.logout();
+}
+
+function revert_settings(pdf_type) {
+    cy.visit(`/courses/${getCurrentSemester()}/sample`);
+    cy.login('instructor');
+    cy.get(`a[href*="/sample/gradeable/${pdf_type}/update"]`).click();
+    cy.get('#page_3_nav').click();
+    cy.get('@minimumGradingGroup').then(setting => {
+        cy.get('#minimum_grading_group').select(setting);
+        cy.get('#minimum_grading_group option:selected').should('have.text', setting);
+    });
+    cy.get('@graderAssignmentMethod').check();
+}
+
+function pdf_access(user_id, tr_number, td_number, gradeable_id, pdf_name) {
+    cy.visit(`/courses/${getCurrentSemester()}/sample`);
+    cy.login(user_id);
+    cy.get(`a[href*="/sample/gradeable/${gradeable_id}/grading/details"]`).click();
+    cy.get('#agree-button').click({ force: true });
+    if (user_id !== 'student') {
+        cy.get(`a[href*='/sample/gradeable/${gradeable_id}/grading/status']`).click();
+        cy.contains('Grading Index').should('be.visible').click();
+    }
+    cy.get('#details-table').should('be.visible');
+    if (user_id !== 'student') {
+        cy.get('.markers-container .btn-default').each(($el) => {
+            if ($el.text().trim() === 'View All') {
+                cy.wrap($el).click();
+            }
+        });
+    }
+    cy.get(`#details-table > tbody.details-content.panel-content-active > tr:nth-child(${tr_number}) > td:nth-child(${td_number}) > a`).eq(0).click();
+    cy.wait(500); //wait for #submission_browser to get loaded
+    cy.get('#submission_browser_btn').then(($el) => {
+        if (!$el.hasClass( 'active' )) {
+            cy.log($el);
+            cy.get($el).find('button').click();
+        }
+    });
+    cy.get('#submissions').click();
+    cy.get('#div_viewer_sd1').find(`a[file-url*='${pdf_name}']`).click();
+    cy.get('#pageContainer1').should('be.visible');
+    cy.get('a[onclick*="collapseFile"]').click();
+    cy.logout();
+}
+
+describe('Test cases for PDFs access', () => {
+    it('users should have access to basic pdfs', () => {
+        const pdf_type = 'grading_homework_pdf';
+        switch_settings('Limited Access Grader', pdf_type);
```

Comment:
`switch_settings()` appears to be used at the start of every test case here.  It would be better to move it to a `.beforeEach()` function to reduce the amount of code duplication.  Apply throughout.